crates/markdown_preview/src/markdown_parser.rs

```diff
@@ -826,11 +825,64 @@ impl<'a> MarkdownParser<'a> {
                     if let Some(image) = self.extract_image(source_range, attrs) {
                         elements.push(ParsedMarkdownElement::Image(image));
                     }
+                } else if local_name!("p") == name.local {
```

Comment:
This only supports the `<p>` tag, but the actual situation is more complex. Text paragraphs can exist within different types, such as `<blockquote>`, `<li>`, even the paragraph text may inside the `<ul>` .

It's difficult for me to clearly describe the problem in simple reply, and I think you doing it implement one by one would take a long time. 

I suggest you can consider covering all types of HTML tags (block and inline) at once.

And using a slightly more complex HTML file as a test case for verification, this will be completed quickly.

And then in HTML Node of the Markdown AST, we just use the `parse_html` to get sub `ParsedMarkdownElement`.

https://github.com/longbridge/gpui-component/blob/f4a292d9757406a6465e17c92d387a0adf8ec8bf/crates/ui/src/text/format/html.rs#L759

Reply:
👌🏼